students/Ben_freeman/lesson03/mailroom.py

```diff
@@ -1,52 +1,51 @@
-"Vladamir": [13.14,1516.17],
-"Santa": [181920.21,22,23.24],
-}
+donor_list=[["Ahmed",123,456,789],
+["Sally", 10,11],
+["Bob", 12],
+["Vladamir", 13.14,1516.17],
+["Santa", 181920.21,22,23.24],
+]
 
 #send thanks funct
-def sndthx():
+def send_thanks():
     while True:
-        qry=input("Please enter a Full Name or enter 'list' to see a list of current donors: ")
-        if qry in donorlist:
-            Newdonate=float(input("How much would you like to donate: \n"))
-            donorlist[qry].append(Newdonate)
-            print(f"\nThank you {qry} for your donation of ${Newdonate}, your donations make the work of the 'American society for taking donations' possible.\n\n"
-                "Sincearly,\n\n"
-                "A low paid intern\n")
-            break
-        elif qry=="list":
-            print('\n'.join(donorlist))
-        elif qry not in donorlist:
-            donorlist[qry]=[]
-            Newdonate=float(input("\nHow much would you like to donate: \n"))
-            donorlist[qry].append(Newdonate)
-            print(f"\nThank you {qry} for your donation of ${Newdonate}, your donations make the work of the 'American society for taking donations' possible.\n\n"
-                "Sincearly,\n\n"
-                "A low paid intern\n")
-            break
+        query=input("Please enter a Full Name, enter 'list' to see a list of current donors or type quit to exit: ")
+        for i in range(0,len(donor_list)):
+            if query in donor_list[i][0]:
+                new_donate=float(input("How much would you like to donate: \n"))
+                donor_list[i].append(new_donate)
+                return print(f"\nThank you {query} for your donation of ${new_donate}, your donations make the work of the 'American society for taking donations' possible.\n\n"
+                    "Sincearly,\n\n"
+                    "A low paid intern\n")
+                
+            elif query=="list":
+                    print(donor_list[i][0])
+            elif query.lower()=="quit":
+                return
+            elif query not in donor_list[i][0]:
+                new_donate=float(input("\nHow much would you like to donate: \n"))
+                donor_list.append([str(query),new_donate])
+                return print(f"\nThank you {query} for your donation of ${new_donate}, your donations make the work of the 'American society for taking donations' possible.\n\n"
+                    "Sincearly,\n\n"
+                    "A low paid intern\n")
+            
+                
+
 
-#making a sortable list
-def sdonor(dlist):
-    mrgl={}
-    for i in dlist:
-        mrgl[i]=[sum(dlist[i]),len(dlist[i]),sum(dlist[i])/len(dlist[i])]
-    flist=[]
-    for keys, values in mrgl.items():
-        values.append(keys)
-        flist.append(values)
-    return sorted(flist, key=lambda flist:flist[0], reverse=True)
+#making a sorted list
+def sorting_function(dlist):
+    sorted_list=[]
+    for i in range(0,len(dlist)):
```

Comment:
you should loop over the list item directly instead of using `range()` - there are uses for this but not very common.

```
for donor in dlist:
    
```

Reply:
Oh, that's much nicer looking and more straight forward, thank you!